eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -127,6 +128,33 @@ class Router(val nodeParams: NodeParams, watcher: ActorRef, initialized: Option[
       sender ! GetNetworkStatsResponse(d.stats)
       stay
 
+    case Event(GetRoutingStateStreaming, d) =>
+      val listener = sender
+      d.nodes
+        .values
+        .sliding(100, 100)
+        .map(NodesDiscovered)
+        .foreach(listener ! _)
+      d.channels
+        .values
+        .map(pc => SingleChannelDiscovered(pc.ann, pc.capacity, pc.update_1_opt, pc.update_2_opt))
+        .sliding(100, 100)
+        .map(ChannelsDiscovered)
+        .foreach(listener ! _)
+      listener ! RoutingStateStreamingUpToDate
+      context.actorOf(Props(new Actor with ActorLogging {
+        log.info(s"subscribing listener=$listener to network events")
+        context.system.eventStream.subscribe(listener, classOf[NetworkEvent])
+        context.watch(listener)
```

Comment:
You are right, although it is a tiny memory leak (front nodes should be long lived). Will look into it...

Reply:
I verified that, in untrusted mode, death watches are still triggered when the whole actor system goes down (as opposed to: when the specific actor that is watched dies). It's good, because here the lifespan of the watched actor is the same as the actor system, so there is no memory leak.

We do have a nasty race condition here though. When we restart the front, the death watch may trigger immediately because of a race with the cluster membership lifecycle (see https://github.com/akka/akka/blob/93a69c42ff465603504fa32f4cb91070618676ce/akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala#L174). The subscription to `NetworkEvent`s will be cancelled right after it is set. This has nasty side effects, because we rely on those `NetworkEvent`s to acknowledge messages from our peers on the front. This resulted in stuck tcp connections.

The fix is in two steps:
- d4191cd53 or 29a1c38bb (competing commits, the latter partially reverts the former) make the message acknowledgement independent of the `NetworkEvent`s. This is a bit complicated because there are several potential race conditions to take into account. See the comment in the `GossipDecision.Accepted` handler of the `FrontRouter` for more details.
- 1d800e8b4 and 8971d9725 solve the death watch issue by adding some time for the cluster to properly detect the new incarnation of the same front server when we restart.